gateway/ln-gateway/src/actor.rs

```diff
@@ -18,14 +19,15 @@ pub struct GatewayActor {
 }
 
 impl GatewayActor {
-    pub async fn new(client: Arc<GatewayClient>) -> Result<Self> {
+    pub async fn new(client: Arc<GatewayClient>, route_hints: Vec<RouteHint>) -> Result<Self> {
         // Retry gateway registration
+        let gateway_registration = client.config().to_gateway_registration_info(route_hints);
```

Comment:
So, we calculate the hint from the CLN at runtime and it will update every time federation is joined or the gateway is restarted? Nice.